pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -1422,3 +1424,175 @@ func TestGetProjectFromKube(t *testing.T) {
 		})
 	}
 }
+
+func TestChangeDeploymentType(t *testing.T) {
+	tests := map[string]struct {
+		deployment *akov2.AtlasDeployment
+	}{
+		"should fail when existing cluster is regular but manifest defines a serverless instance": {
```

Comment:
I was expecting two inputs; existing and new spec. How can this work with one input only?

Reply:
The input is an already-created deployment that will conflict with its mocked counterpart (lines 1534-1560) coming from Atlas.

Reply:
Fine. I'd rather make the test logic dumb simple and ensure the inputs are more explicit, but will not block the PR on this.